File: sesac7_flutter_recipe_app/lib/data/repository/recipe_repository/recipe_repository.dart
import 'package:flutter_recipe_app/core/enum/network_error.dart';
import 'package:flutter_recipe_app/core/result.dart';
import 'package:flutter_recipe_app/data/model/recipe.dart';

abstract interface class RecipeRepository {
  Future<Result<List<Recipe>, NetworkError>> getRecipes();
  Future<Result<List<Recipe>, NetworkError>> searchRecipes(String? keyword);
  Future<Result<List<Recipe>, NetworkError>> getRecentRecipes();
}

File: compose_recipe/app/src/main/java/com/example/recipeapp/data/repository/RecipeRepository.kt
package com.example.recipeapp.data.repository

import com.example.recipeapp.core.Result
import com.example.recipeapp.core.enum.NetworkErrorType
import com.example.recipeapp.data.model.Recipe

interface RecipeRepository {
    suspend fun getRecipes(): Result<List<Recipe>, NetworkErrorType>
    suspend fun findRecipeById(id: Int): Result<Recipe, NetworkErrorType>
    suspend fun getSavedRecipes(): Result<List<Recipe>, NetworkErrorType>
}